modules/ding_provider/ding_provider.module

```diff
@@ -763,3 +764,36 @@ function ding_provider_entity_info_alter(&$entity_info) {
   $entity_info['profile2']['entity cache'] = FALSE;
   $entity_info['profile2_type']['entity cache'] = FALSE;
 }
+
+/**
+ * Implements hook_cron().
+ */
+function ding_provider_cron() {
+  $last_cron_time = variable_get('ding_provider_last_cron', 0);
+
+  // Run daily.
+  if ($last_cron_time < REQUEST_TIME - 86400) {
+    // Remove inactive users.
+    $inactivity_threshold = variable_get('wipe_provider_users_threshold', 0);
+
+    if ($inactivity_threshold) {
+      $query = db_select('users', 'u');
+      $query->join('authmap', 'a', 'u.uid = a.uid');
+      $query
+        ->fields('u', array('uid'))
+        ->condition('u.access', REQUEST_TIME - $inactivity_threshold, '<')
+        ->range(0, 500);
+
+      $result = $query
+        ->execute()
+        ->fetchCol();
+
+      if (!empty($result)) {
+        // TODO: Maybe rework as a queue.
```

Comment:
I think that would be a very good idea to implement this as a queue it might be a very large batch that have to be deleted.